src/features/tokens/types.ts

```diff
@@ -1,22 +1,83 @@
+import { ethers } from 'ethers';
 import { z } from 'zod';
 
-/**
- * Zod schema for Token config validation
- */
-export const TokenSchema = z.object({
-  chainId: z.number(),
-  address: z.string(),
-  hypCollateralAddress: z.string(),
+import { TokenType } from '@hyperlane-xyz/hyperlane-token';
+
+const commonTokenFields = z.object({
+  chainId: z.number().positive(),
   name: z.string(),
   symbol: z.string(),
   decimals: z.number().positive(),
   logoURI: z.string().optional(),
 });
+type CommonTokenFields = z.infer<typeof commonTokenFields>;
+
+/**
+ * Types for the developer-provided config
+ * See src/consts/tokens.ts
+ */
+
+interface BaseTokenConfig extends CommonTokenFields {
+  type: `${TokenType}`; // use template literal to allow string values
+}
+
+const CollateralTokenSchema = commonTokenFields.extend({
+  type: z.literal(TokenType.collateral),
+  address: z.string(),
+  hypCollateralAddress: z.string(),
+});
 
-export const TokenListSchema = z.array(TokenSchema);
+interface CollateralTokenConfig extends BaseTokenConfig {
+  type: TokenType.collateral | 'collateral';
```

Comment:
I'm surprised we need to do this if the enum assigns `TokenType.collateral` to 'collateral'. Does it?

Reply:
Typescript won't allow a field of type `TokenType` to be the literal value 'collateral' even if the enum's values match up. And I was intent on letting devs use the literals in the configs to avoid needing to import `TokenType`. This separation of *Config and *Metadata types is the solution